qiskit/primitives/containers/bit_array.py

```diff
@@ -0,0 +1,345 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+BitArray
+
+Details
+=======
+
+    We use the word "samples" in reference to the fact that this is a primary data type returned by
+    the Sampler whose job is to supply samples, and whose name we can't easily change.
+    This is certainly confusing because this library uses the word "samples" to also refer to random
+    circuit instances.
```

Comment:
This was copied from some private prototype code. We probably shouldn't claim certain confusion in the header :P

But it does raise an important point: SamplerV2 calls them shots, but this class calls them samples. We should unify, likely to "shots". This means changing the `@property def samples` etc.

Reply:
Should we remove BitArray from this PR and work on it in a separate one? Its not needed for BaseSamplerV2 or SamplerPub

Reply:
I removed BitArray, so can @ihincks or @t-imamichi open a new PR adding that class and its tests?

Reply:
I'm on it.

Reply:
#11542